src/chain/mod.rs

```diff
@@ -538,6 +679,118 @@ impl ChainSource {
 
 				res
 			},
+			Self::Electrum {
+				electrum_runtime_client,
+				onchain_wallet,
+				onchain_wallet_sync_status,
+				kv_store,
+				logger,
+				node_metrics,
+				..
+			} => {
+				let electrum_client: Arc<ElectrumRuntimeClient> =
+					if let Some(client) = electrum_runtime_client.read().unwrap().as_ref() {
+						Arc::clone(client)
+					} else {
+						debug_assert!(
+							false,
+							"We should have started the chain source before syncing the onchain wallet"
+						);
+						return Err(Error::FeerateEstimationUpdateFailed);
+					};
+				let receiver_res = {
```

Comment:
This code block including the if below seems to be identical across all backends, and backend independent. Maybe ok for now, but avoiding code duplication may be better.